Utilities/UtilData.cs

```diff
@@ -1,156 +1,201 @@
+        public static string ToString(byte[] Input)
+        {
+            if (Input == null || Input.Length == 0 || Input.Length % 2 == 1) { throw new FormatException("Given byte[] does not convert to string."); }
+            StringBuilder Output = new StringBuilder(Input.Length / 2);
+            for(int i = 0; i < Input.Length; i += 2)
+            {
+                Output.Append((char)(Input[i] << 8 | Input[i + 1]));
+            }
+            return Output.ToString();
+        }
+
+        /// <summary> Tries to convert a byte input (unicode) to a string </summary>
+        /// <param name="Input"> Bytes to convert to string </param>
+        /// <param name="Output"> Output of the conversion. (Null if failed) </param>
+        /// <returns> True if string conversion succeeds </returns>
+        public static bool TryToString(byte[] Input, out string Output)
+        {
+            Output = null; 
+            try { Output = ToString(Input); }
+            catch { return false; }
+            return true;
+        }
+
+        /// <summary> Determines if the given type is numeric. </summary>
+        /// <param name="Type"> Type to determine whether or not it is a numeric </param>
+        /// <returns> Returns <c>true</c> if param is a numeric; otherwise returns <c>false</c>. </returns>
+        public static bool IsNumericType(Type Type)
+        {
+            switch (Type.GetTypeCode(Type))
+            {
+                case TypeCode.Byte:
+                case TypeCode.SByte:
+                case TypeCode.UInt16:
+                case TypeCode.UInt32:
+                case TypeCode.UInt64:
+                case TypeCode.Int16:
+                case TypeCode.Int32:
+                case TypeCode.Int64:
+                case TypeCode.Decimal:
+                case TypeCode.Double:
+                case TypeCode.Single:
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        /// <summary> Takes a 16bit number, and swaps the locations of the first and last 8b. E.g. 0x54EC would become 0xEC54. </summary>
+        /// Intended for use with 16b I2C devices that expect the byte order reversed.
+        public static ushort SwapBytes(ushort Input)
+        {
+            return (ushort)(((Input & 0b1111_1111) << 8) | ((Input >> 8) & 0b1111_1111));
+        }
+
+        /// <summary> Computes a CRC-16 checksum from given payload. </summary>
+        /// <param name="payload"> Payload of which to compute the checksum </param>
+        public static ushort CRC16(byte[] payload)
+        {
+            int i;
+            int cksum = 0;
+
+            for (i = 0; i < payload.Length; i++)
+            {
+                cksum = crc16_table[(((cksum >> 8) ^ payload[i]) & 0xFF)] ^ (cksum << 8);
+            }
+
+            return (ushort)cksum;
+
+        }
+
+        private static ushort[] crc16_table = { 0x0000, 0x1021, 0x2042, 0x3063, 0x4084,

```

Comment:
The motor controller documentation is unclear which of the numerous CRC16 checksum types it is using therefore I translated their method of producing the checksum which used this table!

Reply:
🤦‍♂️ I guess I can't blame you for having nothing decent to work from. This will do for now, but it might be worthwhile going back and determining the algorithm used, and replacing this later on.